sematic/db/queries.py

```diff
@@ -158,22 +158,24 @@ def get_run_status_details(
     jobs for the run (if any exist)
     """
     with db().get_session() as session:
-        query_results = (
-            session.query(Run.id, Run.future_state, Run.external_jobs_json)
+        query_results = list(
+            session.query(Run.id, Run.future_state, Job)
+            .outerjoin(Job, Job.run_id == Run.id, full=True)
             .filter(Run.id.in_(run_ids))
+            .filter(sqlalchemy.or_(Job.kind == JobKind.run, Job.kind.is_(None)))
```

Comment:
Is this filter necessary? Would `.outerjoin(Job, Job.run_id == Run.id, full=True)` ever result in a `Job.kind` that isn't a "run"?

My understanding is that "run" -> func, and "resolver" -> driver. Is that correct?

Reply:
The final returned structure is:
```
{
    "run_id_1": (FutureState1, [job1, job2]),
    "run_id_2": (FutureState2, []),
}
```

In order to get things like the second entry in that dict, we need to account for having rows in the returned results which have only a run id and future state and no job. This is explained in the comment I just added above.